smoke/tests/tool/snapshotter.go

```diff
@@ -64,6 +73,34 @@ func NewSnapshotterClient(sock string) *SnapshotterClient {
 	}
 }
 
+func (cli *SnapshotterClient) request(method, urlSuffix string, body any) (respBody []byte, err error) {
+	var reqJSON []byte
+	if body != nil {
+		reqJSON, err = json.Marshal(body)
+		if err != nil {
+			return nil, errors.Wrap(err, "marshal request body")
+		}
+	} else {
+		reqJSON = make([]byte, 0)
+	}
+
+	url := fmt.Sprintf("http://unix%s", urlSuffix)
+	req, err := http.NewRequest(method, url, bytes.NewBuffer(reqJSON))
+	if err != nil {
+		return nil, errors.Wrap(err, "build request")
+	}
+	resp, err := cli.client.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "do request")
+	}
```

Comment:
Should move `defer resp.Body.Close()` to here, and print the possible error message from `io.ReadAll(resp.Body)` if we encounter the abnormal status code except 2xx.